agent/engine/docker_client.go

```diff
@@ -83,6 +83,10 @@ const (
 	// around a docker bug which sometimes results in pulls not progressing.
 	dockerPullBeginTimeout = 5 * time.Minute
 
+	// dockerPullInactivityTimeout is the amount of time that we will
+	// wait when the pulling does not progress
+	dockerPullInactivityTimeout = 5 * time.Minute
```

Comment:
how did we decide on 5 mins? is there any data/metrics available for us? Ideally p99 would be a good value to use.

Reply:
currently it's set arbitrarily. i'm not sure if there's any data/metrics for this

Reply:
hm we havent figured out a good way to measure p99 in this case. the only real visibility we've had so far is from root causing hanging tasks, containers which will get stuck for arbitrary periods of time. we want to introduce as a fail fast for docker inactivity at least during image pulls.

do you know if we could get a better idea of what to set here from ecr side? 😕

Reply:
From ECR side, P99 is in seconds not minutes (I can show you the dashboard in person) so 5 mins is more than long enough timeout. Thought 5 mins was a bit too long in general but if it's only meant to be used for hanging tasks I suppose it is not a big issue. However flip side is whether this might adversely impact large image pull that can take longer than 5 mins? Is ECR the only repo we pull from?

Reply:
>However flip side is whether this might adversely impact large image pull that can take longer than 5 mins?

The `dockerPullInactivityTimeout ` shouldn't adversely affect large image pulls. This is explicitly to checking to timeout if docker is hanging during the download and large image pulls should show continuous activity during the download. 

> Is ECR the only repo we pull from?

No, the agent could be pulling from anywhere.

Reply:
thanks for the explanation! if timeout is specifically for the case where docker is hanging and inactive, then I don't think any of my concerns are relevant.

Reply:
just as a check, i tested it manually with a large image (3GB) and small timeout (5 secs), and the pulling worked fine with no timeout.

Reply:
`5m` seems too long here. I don't think this should be greater than `1m`.

Reply:
The timeout is set to 1 minute now. I have commented on the SageMaker ticket to see if there's any input at their side.